src/main/kotlin/io/ionic/libs/ionfilesystemlib/helper/IONFILEContentHelper.kt

```diff
@@ -0,0 +1,227 @@
+            } ?: throw IONFILEExceptions.UnknownError()
+            return@runCatching fileContents
+        }.mapError()
+    }
+
+    /**
+     * Reads the contents of a file in chunks
+     *
+     * Useful when the file does not fit in entirely memory.
+     *
+     * @param uri the content uri to read the file from
+     * @param options options for reading the file in chunks; refer to [IONFILEReadInChunksOptions]
+     * @return a (cold) flow in which the chunks are emitted;
+     * the flow completes after all chunks are emitted (unless an error occurs somewhere in-between)
+     */
+    fun readFileInChunks(
+        uri: Uri,
+        options: IONFILEReadInChunksOptions
+    ): Flow<String> = flow {
+        contentResolver.openInputStream(uri)?.use { inputStream ->
+            inputStream.readByChunks(
+                options,
+                DEFAULT_BUFFER_SIZE,
+                onChunkRead = { chunk -> emit(chunk) }
+            )
+        } ?: throw IONFILEExceptions.UnknownError()
+    }.catch { throw it.mapError() }
+        .flowOn(Dispatchers.IO)
+
+    /**
+     * Gets information about a file using content resolver
+     *
+     * @param uri the content uri to get the metadata from
+     * @return success with result containing relevant file information, error otherwise
+     */
+    suspend fun getFileMetadata(uri: Uri): Result<IONFILEMetadataResult> =
+        withContext(Dispatchers.IO) {
+            runCatching {
+                val cursor = contentResolver.query(uri, null, null, null, null)
+                    ?: throw IONFILEExceptions.UnknownError()
+                cursor.use {
+                    if (!cursor.moveToFirst()) {
+                        throw IONFILEExceptions.DoesNotExist()
+                    }
+                    contentResolver.persistedUriPermissions
+                    val name = getNameForContentUri(cursor)
+                    val size: Long = getSizeForContentUri(cursor, uri)
+                    val lastModified: Long = getLastModifiedTimestampForContentUri(cursor)
+                    val created: Long = getCreatedTimestampForContentUri(cursor)
+                    val mimeType = contentResolver.getType(uri) ?: FILE_MIME_TYPE_FALLBACK
+                    IONFILEMetadataResult(
+                        fullPath = uri.path ?: "",
+                        name = name,
+                        uri = uri,
+                        size = size,
+                        type = IONFILEFileType.File(mimeType),
+                        createdTimestamp = created,
+                        lastModifiedTimestamp = lastModified
+                    )
+                }
+            }.mapError()
+        }
+
+    /**
+     * Deletes a file using content resolver
+     *
+     * @param uri the content uri of the file to delete
+     * @return success if file was deleted, error otherwise
+     */
+    suspend fun deleteFile(uri: Uri): Result<Unit> = withContext(Dispatchers.IO) {
```

Comment:
This could be simplified to the following right?

```
suspend fun deleteFile(uri: Uri): Result<Unit> = withContext(Dispatchers.IO) {
    runCatching {
        val rowsDeleted = contentResolver.delete(uri, null, null)
        if (rowsDeleted <= 0) {
            throw IONFILEExceptions.UnknownError()
        }
    }.mapError()
}
```